[Improvement] Improve use of JavaUtilDate in KerberosTokenProvider.java

### What would you like to be improved?

KerberosTokenProvider.java:122: warning: [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.

```java
return tickets.iterator().next().getEndTime().getTime() < System.currentTimeMillis();
```

See https://errorprone.info/bugpattern/JavaUtilDate


### How should we improve?

Consider improving this

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -259,6 +259,7 @@ subprojects {
         "InvalidJavaTimeConstant",
         "InvalidPatternSyntax",
         "IsInstanceIncompatibleType",
+        "JavaUtilDate",
         "JUnit4ClassAnnotationNonStatic",
         "JUnit4SetUpNotRun",
         "JUnit4TearDownNotRun",

--- a/clients/client-java/src/main/java/com/datastrato/gravitino/client/KerberosTokenProvider.java
+++ b/clients/client-java/src/main/java/com/datastrato/gravitino/client/KerberosTokenProvider.java
@@ -113,6 +113,7 @@ public final class KerberosTokenProvider implements AuthDataProvider {
         });
   }
 
+  @SuppressWarnings("JavaUtilDate")
   private boolean isLoginTicketExpired() {
     Set<KerberosTicket> tickets =
         loginContext.getSubject().getPrivateCredentials(KerberosTicket.class);

--- a/clients/client-java/src/test/java/com/datastrato/gravitino/client/TestOAuth2TokenProvider.java
+++ b/clients/client-java/src/test/java/com/datastrato/gravitino/client/TestOAuth2TokenProvider.java
@@ -31,6 +31,7 @@ import org.mockserver.integration.ClientAndServer;
 import org.mockserver.matchers.Times;
 import org.mockserver.model.HttpResponse;
 
+@SuppressWarnings("JavaUtilDate")
 public class TestOAuth2TokenProvider {
 
   private static final int PORT = 1082;

--- a/integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/rest/OAuth2OperationsIT.java
+++ b/integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/rest/OAuth2OperationsIT.java
@@ -33,6 +33,7 @@ public class OAuth2OperationsIT extends AbstractIT {
 
   private static String token;
 
+  @SuppressWarnings("JavaUtilDate")
   @BeforeAll
   public static void startIntegrationTest() throws Exception {
     Map<String, String> configs = Maps.newHashMap();

--- a/server-common/src/test/java/com/datastrato/gravitino/server/auth/TestOAuth2TokenAuthenticator.java
+++ b/server-common/src/test/java/com/datastrato/gravitino/server/auth/TestOAuth2TokenAuthenticator.java
@@ -22,6 +22,7 @@ import java.util.Map;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+@SuppressWarnings("JavaUtilDate")
 public class TestOAuth2TokenAuthenticator {
 
   @Test

